test_pull_requests

```diff
@@ -1189,8 +1388,43 @@ popd
         # comment to reflect the new state
         if submit_test_job
           delete_comment_with_prefix(id, base_repo, ACTION_PREFIX, comments)
-          extended_tests = get_extended_tests(req, comments, branch, settings)
-          submit_tests(repo_to_pull_request, base_repo, branch, comment_id, extended_tests, settings)
+          # Check for pretest_settings_key, so we might skip a round
+          # of tests prior to merge
+          pretest_settings = Properties['settings'][settings['pretest_settings_key']]
+          repo_to_comment_fields = {}
+          can_skip_merge_tests = false
+          if pretest_settings && repo_to_pull_request.length > 0
+            pretest_prefix = pretest_settings['test_prefix']
+            can_skip_merge_tests = repo_to_pull_request.all? do |repo, pull_request|
+              pretest_comment = get_comment_with_prefix(pull_request['number'], repo, pretest_prefix, (repo == base_repo ? comments : nil))
+              if pretest_comment.nil?
+                false
+              else
+                pretest_fields = extract_bot_comment_fields(pretest_comment['body'], pretest_settings)
+                can_skip_one = (pretest_fields[:state] == :success &&
+                                pretest_fields[:base_commit] == pull_request['base']['sha'])
+                if can_skip_one
+                  # Store pre-merge carry the build url and base commit id through to SUCCESS
+                  repo_to_comment_fields[repo] = {:state => :success, :build_url => pretest_fields[:build_url], :base_commit => pretest_fields[:base_commit]}
+                end
+                can_skip_one
+              end
+            end
+          end
+          if can_skip_merge_tests
+            $stderr.puts "Skipping #{settings['name']} tests for pull request ##{id} for repo '#{base_repo}'"
+            repo_to_pull_request.each do |repo, pull_request|
+              test_merge_pull_request(pull_request['number'], repo, settings)
+            end
+            repo_to_pull_request.each do |repo, pull_request|
+              $stderr.puts "\n*******Merging tested, up-to-date pull request: #{GITHUB_BASE_URL}/#{Properties['github_user']}/#{repo}/pull/#{pull_request['number']} "
+              merge_pull_request(pull_request['number'], repo, settings, nil, repo_to_comment_fields[repo])
```

Comment:
Which I know you don't like, but I think Dan's arguments in favor are convincing. :D